src/api/app/models/workflow/step/set_flags.rb

```diff
@@ -0,0 +1,82 @@
+class Workflow::Step::SetFlags < ::Workflow::Step
+  REQUIRED_KEYS = [:flags].freeze
+  REQUIRED_FLAG_KEYS = [:type, :status, :project].freeze
+  OPTIONAL_FLAG_KEYS = [:package, :repository, :architecture].freeze
+
+  validate :validate_flags
+
+  def call(_options = {})
+    return unless valid?
+
+    set_flags
+  end
+
+  def flags
+    step_instructions[:flags]
+  end
+
+  private
+
+  def set_flags
+    ActiveRecord::Base.transaction do
+      flags.each do |flag|
+        project = Project.find_by(name: target_project_name(project_name: flag[:project]))
+        raise ActiveRecord::RecordNotFound, "Couldn't find Project #{flag[:project]}" if project.nil?
```

Comment:
if you are doing a `find_by` and raising `ActiveRecord::RecordNotFound`, couldn't you use `find_by!` ?

Reply:
Good point. It's updated.